catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/adapter/BrickAdapter.kt

```diff
@@ -273,27 +282,41 @@ class BrickAdapter(private val sprite: Sprite) :
         for (item in items) {
             val brickPosition = items.indexOf(item)
             viewStateManager.setEnabled(
-                selectableForCopy(brickPosition, scriptSelected),
-                brickPosition
+                selectableForCopy(brickPosition, scriptSelected), brickPosition
             )
         }
     }
 
+    private fun isChildBrickOfSelectedCompositeBrick(brickPosition: Int): Boolean {
+        val childBrick = items[brickPosition]
+        if (!selectedItems.contains(childBrick.parent)) {
+            return false
+        }
+        if (childBrick.parent is IfLogicBeginBrick.ElseBrick || childBrick.parent is PhiroIfLogicBeginBrick.ElseBrick) {
+            return true
+        }
+        if (childBrick.parent !is CompositeBrick) {
+            return false
+        }
+        if ((childBrick.parent as CompositeBrick).nestedBricks?.contains(childBrick) == true) {
+            return true
+        }
+        if ((childBrick.parent as CompositeBrick).secondaryNestedBricks?.contains(childBrick) == true) {
+            return true
+        }
+        return false
+    }
```

Comment:
still really bad structure of function, what is the purpose of your first if (!selecetedItems.contains...)?
```suggestion
    private fun isChildBrickOfSelectedCompositeBrick(brickPosition: Int): Boolean {
        val childBrick = items[brickPosition]
        val parent = childBrick.parent
        return (parent is IfLogicBeginBrick.ElseBrick || parent is PhiroIfLogicBeginBrick.ElseBrick ||
            parent is CompositeBrick && checkNestedBricks(parent, childBrick))
    }

    private fun checkNestedBricks(parent: CompositeBrick, childBrick: Brick): Boolean {
        return (parent.nestedBricks.contains(childBrick) ||
            parent.hasSecondaryList() && parent.secondaryNestedBricks.contains(childBrick))
    }
```
i've tried the tests with this and everything works fine, was the first check therefore redundant? try in general to not overuse if(condition){return false/true}... in general checking for something to return false is a bad habit the suggestion is much cleaner and also not that confusing

Reply:
The first check (selectedItems.contains(childBrick.parent)) is necessary. Without it the tests doesnt run through.